code/script.js

```diff
@@ -218,108 +221,145 @@ const generateBoard = () => {
+        person[category] === value
+      );
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
-  }
-
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+      alert (
+        `No, the person doesn't have ${value} ${category}! Remove all people that has ${value} ${category}`
+      );
+      charactersInPlay = charactersInPlay.filter((person) => 
+        person[category] !== value
+      );
+    };
+  };
+  generateBoard();
+};
 
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+// when clicking guess, the player have to confirm that they want to make a guess.
+const guess = (personToConfirm) => { 
+  let userConfirm = confirm(`Do you think the secret person is ${personToConfirm}?`)
+  if (userConfirm) {
+    checkMyGuess(personToConfirm)
+  };
+};
 
-  // Invoke a function to redraw the board with the remaining people.
-}
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
-const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
-
-// If you confirm, this function is invoked
+// If player confirm, this function is invoked
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
-}
+  if (personToCheck === secret.name) {
+    winOrLoseText.innerHTML = `You guessed ${numberOfGuesses} times and got it right, ${personToCheck} is the secret person!! Good job!`
+  } else {
+    winOrLoseText.innerHTML = `Ooh noo! You guessed ${numberOfGuesses} times but got it wrong :( ${personToCheck} is not the secret person, ${secret.name} is!`
+  };
+  winOrLose.style.display = 'flex';
+};
 
 // Invokes the start function when website is loaded
-start()
+start();
 
 // All the event listeners
-restartButton.addEventListener('click', start)
+restartButton.addEventListener('click', () => {
+  location.reload();
+  start();
+});
+questions.addEventListener('change', selectQuestion);
+
+findOutButton.addEventListener('click', () => { 
+  checkQuestion();
+  numberOfGuesses += 1;
+  guessCounter.innerHTML = `${numberOfGuesses}`;
+});
+playAgainButton.addEventListener('click', () => {
+  start ();
```

Comment:
If you would like the counter to restart itself when you play again, add `numberOfGuesses = 0
    guessCounter.innerHTML =`${numberOfGuesses}`;` on this row after start()